### PR TITLE
APP-3071: Update the configuration field for privateKey and certificate

### PR DESCRIPTION
**Pull request by symphony-hong**
### Ticket
[APP-3071](https://perzoinc.atlassian.net/browse/APP-3071)

### Description
Update the configuration field for privateKey and certificate

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Javadoc added or updated
- [x] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -118,20 +120,31 @@ Required `config.yaml` setup:
 host: acme.symphony.com
 app:
     appId: app-id
-    privateKeyPath: /path/to/rsa/private-key.pem
+    privateKey:
+      path: /path/to/rsa/private-key.pem
 ```
 
+> The private key path can still be configured under `privateKeyPath` field but will be deprecated in the future:
```
Comment (symphony-youri) [inline docs/authentication.md:127]: I would perhaps not even mention it to keep it simple

**Review by symphony-youri — COMMENTED** [review]
I found traces of privateKeyPath in configuration.md app-starter and core-starter
and in some of the examples/tests configuration files
should we update those too?

**Comment by symphony-youri** [comment]
Accordingly should we update trustStorePath in BdkSslConfig too?
proxyUrl / proxyUsername / proxyPassword in BdkClientConfig

**Comment by symphony-hong** [comment]
> Accordingly should we update trustStorePath in BdkSslConfig too?
> proxyUrl / proxyUsername / proxyPassword in BdkClientConfig

@symphony-youri I would prefer to have it in PR for proxy ticket. @symphony-elias WDYT?

**Review by symphony-youri — APPROVED** [review]

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/model/BdkSslConfig.java [hunk 2]
```diff
@@ -11,4 +13,19 @@
 
     private String trustStorePath;
     private String trustStorePassword;
```
Comment (symphony-thibault) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/model/BdkSslConfig.java:15]: We need to `@Deprecated` those fields

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/model/BdkAuthenticationConfig.java [hunk 3]
```diff
@@ -17,41 +17,61 @@
   protected byte[] certificateContent;
   protected String certificatePassword;
```
Comment (symphony-thibault) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/model/BdkAuthenticationConfig.java:18]: We need to `@Deprecated` those fields

**Review by symphony-thibault — APPROVED** [review]

**Pull request merged**